Helper/FilterMapper.php

```diff
@@ -0,0 +1,149 @@
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Cmp\Helper;
+
+use Magento\Catalog\Model\Layer\Filter\Item;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Store\Model\Store;
+use Nosto\Operation\Recommendation\Filters;
+use Nosto\Tagging\Helper\Data as NostoHelperData;
+
+class FilterMapper
+{
+    /** @var Filters */
+    private $filters;
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var string */
+    private $brand;
+
+    /**
+     * FilterMapper constructor.
+     * @param Filters $filters
+     * @param NostoHelperData $nostoHelperData
+     */
+    public function __construct(Filters $filters, NostoHelperData $nostoHelperData)
+    {
+        $this->filters = $filters;
+        $this->nostoHelperData = $nostoHelperData;
+    }
+
+    /**
+     * @param Store $store
+     */
+    public function init(Store $store): void
+    {
+        $this->brand = $this->nostoHelperData->getBrandAttribute($store);
+    }
+
+    /**
+     * @param Item $item
+     * @throws LocalizedException
+     */
+    public function mapFilter(Item $item): void
+    {
+        /** @var string $frontendInput */
+        $frontendInput = $item->getFilter()->getData('attribute_model')
+            ->getData('frontend_input');
+
+        $filterName = $item->getName();
+        $value = '';
+        switch ($frontendInput) {
+            case 'price':
+                $value = $item->getData('value');
+                break;
+            case 'select':
+                $value = $item->getData('label');
+                break;
+            case 'multiselect':
+                $value = $item->getData('label');
+                break;
+            case 'date':
+                break;
+            case 'boolean':
+                $value = $item->getData('value') === '1';
+                break;
+        }
+        $this->setValue($filterName, $value);
+    }
+
+    /**
+     * @param string $name
+     * @param string|array $value
+     */
+    private function setValue(string $name, $value)
+    {
+        if ($this->brand === $name) {
+            $this->filters->setBrands($this->getArray($value));
+            return;
+        }
+
+        switch (strtolower($name)) {
+            case 'price':
+                $this->filters->setPrice(min($value), max($value));
+                break;
+            case 'new':
+                $this->filters->setFresh($value);
+                break;
+            default:
+                $this->filters->setCustomFields($name, $this->getArray($value));
+                break;
+        }
+    }
+
+    /**
+     * @return Filters
+     */
+    public function getFilters(): Filters
+    {
+        return $this->filters;
+    }
+
+    /**
+     * @param string|array $value
+     * @return array
+     */
+    private function getArray($value)
```

Comment:
This method is not a getter from this entity. Also, every getter should have a setter.
I'd rename this to `makeArrayFromValue` or something like this.